deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py

```diff
@@ -0,0 +1,112 @@
+    ----------
+    data_format : str, default: 'TextData'
+        Represent the format of the returned value. Can be 'TextData'|'DataFrame'
+        'TextData' will return the data as a TextData object
+        'Dataframe' will return the data as a pandas DataFrame object
+    as_train_test : bool, default: True
+        If True, the returned data is split into train and test exactly like the toy model
+        was trained. The first return value is the train data and the second is the test data.
+        In order to get this model, call the load_fitted_model() function.
+        Otherwise, returns a single object.
+    use_full_size : bool, default: False
+        If True, the returned data will be the full dataset, otherwise returns a subset of the data.
+
+    Returns
+    -------
+    dataset : Union[TextData, pd.DataFrame]
+        the data object, corresponding to the data_format attribute.
+    train, test : Tuple[Union[TextData, pd.DataFrame],Union[TextData, pd.DataFrame]
+        tuple if as_train_test = True. Tuple of two objects represents the dataset split to train and test sets.
+    """
+    if data_format.lower() not in ['textdata', 'dataframe']:
+        raise ValueError('data_format must be either "Dataset" or "Dataframe"')
+
+    data = read_and_save_data(ASSETS_DIR, 'just_dance_data.csv', _FULL_DATA_URL, to_numpy=False)
+    data['dateComment'] = pd.to_datetime(data['dateComment'])
+
+    if not as_train_test:
+        if not use_full_size:
+            data = data[(data['dateComment'] < '2013-01-01') | (data['dateComment'] >= '2021-01-01')]
+        if data_format.lower() != 'textdata':
+            return data
+
+        label = data.drop(columns=[_TEXT_COL] + _METADATA_COLS).to_numpy().astype(int)
+        dataset = TextData(data[_TEXT_COL], label=label, task_type='text_classification',
+                           metadata=data[_METADATA_COLS], categorical_metadata=_CAT_METADATA)
+        return dataset
+
+    else:
+        if use_full_size:
+            train = data[data['dateComment'] < '2015-01-01']
+            test = data[data['dateComment'] >= '2015-01-01']
+        else:
+            train = data[data['dateComment'] < '2013-01-01']
+            test = data[data['dateComment'] >= '2021-01-01']
+
+        if data_format.lower() != 'textdata':
+            return train, test
+
+        train_metadata, test_metadata = train[_METADATA_COLS], test[_METADATA_COLS]
+        label_train = train.drop(columns=[_TEXT_COL] + _METADATA_COLS).to_numpy().astype(int)
+        label_test = test.drop(columns=[_TEXT_COL] + _METADATA_COLS).to_numpy().astype(int)
+
+        train_ds = TextData(train[_TEXT_COL], label=label_train, task_type='text_classification',
+                            metadata=train_metadata, categorical_metadata=_CAT_METADATA)
+        test_ds = TextData(test[_TEXT_COL], label=label_test, task_type='text_classification',
+                           metadata=test_metadata, categorical_metadata=_CAT_METADATA)
+
```

Comment:
Add future tasks to add saved predictions, properties and embeddings here?

Reply:
will do in next pr